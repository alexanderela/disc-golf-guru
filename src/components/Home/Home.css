.Home {
  align-items: center;
  background-color: #FFF;
  border: 3px solid #febb00;
  border-radius: 20px;
  display: flex;
  flex-direction: column;
  height: 15rem;
  justify-content: center;
  margin: 1.25rem auto;
  padding: 0.75rem 0 1rem 0;
  width: 38rem;
  box-shadow: 0px 0px 1px 2px #Feeb00;
}

.home-header {
  margin: 0 0 1rem;
  font-size: 1.5rem;
}

.home-intro {
  margin: 1rem 4rem;
  line-height: 1.25rem;
}

.home-search-button {
  background-color: #2c9a2c;
  border: none;
  border-radius: 20px;
  color: #FFF;
  cursor: pointer;
  font-size: 1.25rem;
  font-weight: 700;
  height: 2.5rem;
  margin: 1rem 0 0 0.5rem;
  outline-width: 0;
  padding: 0 1.25rem;
  text-shadow: #000 0 0 0;
}

.home-search-instructions {
  font-size: 1.25rem;
  font-weight: 600;
  color: #2c9a2c;
  margin: 1rem 4rem 0;
}

.home-search-button:hover {
  background-color: #febb00;
  color: #524949;
}