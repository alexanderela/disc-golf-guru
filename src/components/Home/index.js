import React from 'react';
import './Home.css';

const Home = props => (
  <div className="Home">
    <h2 className="home-header">Welcome to Disc Golf Guru!</h2>
    <p className="home-intro">
      This site lets you search for nearby disc golf courses and shows your
      current weather and forecasts for each course.
    </p>
    <p className="home-search-instructions">
      Click the 'Find A Course' button above to search for a disc golf course!
    </p>
  </div>
);

export default Home;
