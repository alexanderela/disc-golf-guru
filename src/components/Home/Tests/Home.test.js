import React from 'react';
import ReactDOM from 'react-dom';
import Home from '../';
import { shallow } from 'enzyme';

describe('Home', () => {
	let wrapper;

	beforeEach(() => {
		wrapper = shallow(<Home />)
	})
	it('should render like snapshot', () => {
		expect(wrapper).toMatchSnapshot()
	})
	
})