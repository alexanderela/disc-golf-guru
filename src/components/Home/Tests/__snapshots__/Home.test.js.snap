// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render like snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="home-header"
        >
          Welcome to Disc Golf Guru!
        </h2>,
        <p
          className="home-intro"
        >
          This site lets you search for nearby disc golf courses and shows your current weather and forecasts for each course.
        </p>,
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/findcourses"
        >
          <button
            className="home-search-button"
          >
            Click here to search for a disc golf course
          </button>
        </NavLink>,
      ],
      "className": "Home",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Welcome to Disc Golf Guru!",
          "className": "home-header",
        },
        "ref": null,
        "rendered": "Welcome to Disc Golf Guru!",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "This site lets you search for nearby disc golf courses and shows your current weather and forecasts for each course.",
          "className": "home-intro",
        },
        "ref": null,
        "rendered": "This site lets you search for nearby disc golf courses and shows your current weather and forecasts for each course.",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": <button
            className="home-search-button"
          >
            Click here to search for a disc golf course
          </button>,
          "to": "/findcourses",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Click here to search for a disc golf course",
            "className": "home-search-button",
          },
          "ref": null,
          "rendered": "Click here to search for a disc golf course",
          "type": "button",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="home-header"
          >
            Welcome to Disc Golf Guru!
          </h2>,
          <p
            className="home-intro"
          >
            This site lets you search for nearby disc golf courses and shows your current weather and forecasts for each course.
          </p>,
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/findcourses"
          >
            <button
              className="home-search-button"
            >
              Click here to search for a disc golf course
            </button>
          </NavLink>,
        ],
        "className": "Home",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome to Disc Golf Guru!",
            "className": "home-header",
          },
          "ref": null,
          "rendered": "Welcome to Disc Golf Guru!",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "This site lets you search for nearby disc golf courses and shows your current weather and forecasts for each course.",
            "className": "home-intro",
          },
          "ref": null,
          "rendered": "This site lets you search for nearby disc golf courses and shows your current weather and forecasts for each course.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": <button
              className="home-search-button"
            >
              Click here to search for a disc golf course
            </button>,
            "to": "/findcourses",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Click here to search for a disc golf course",
              "className": "home-search-button",
            },
            "ref": null,
            "rendered": "Click here to search for a disc golf course",
            "type": "button",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
